### PR TITLE
FEAT: Menu Page, and updating url creation function, 

### PR DESCRIPTION
**Pull request by MinniMinMin**
### What's Changed
Home Algorithms Page created.
Search Bar for algorithms implemented.
Left panel in algorithm visualisation pages removed, and implemented Dropdown Algorithms Menu instead.
Fix for share button url creation. URL creation logic is now in new file - 'urlCreator.js' in mid-panel folder. Contains the old switch statement that was previously found in the index file. Index.js now calls the function instead of containing it.
### Screenshots
1.
image
2.
image
3.
image
4.
image
image

### Checklist
Verified on Mac
Verified on Windows

### How to Test
Clone the repository and run npm install.
Start the server and frontend with npm start.

### Vercel Deployment
https://algorithms-in-action-git-2024sem2-melanie-hs-projects.vercel.app/?_vercel_share=5Lv3bBoPWUzwWomXuWn2wUsofrLZCiTi

### Documentation
Fix for URL creation: Logic for creating the url remains the same, updated documentation can be found in:
[docs/URL-Code-Implementation-Guideline-20241011.pdf](https://github.com/algorithms-in-action/algorithms-in-action.github.io/pull/80/files#diff-2f9f160af7dd8d1536635e96dbab10b28d9780a133021cebefb1ebc03bb389b4)

Documentation for the home page can be found in:
[docs/Homepage-Guidelines-20241020.pdf](https://github.com/algorithms-in-action/algorithms-in-action.github.io/pull/80/files#diff-4eef3e3414a83b439c6ad9c476996091b65841ac13c9526d2ca9683cb383dae1)

### CODE REVIEW

**Comment by MaryHuang8** [comment]
@MinniMinMin Please rename the PR title to containing only one action keyword 😄 
Also push the guideline document to under `docs`, and specify the name and location for it. This is because not everyone would have access to our Confluence 🫶.
PR message otherwise looking good!

**Comment by MaryHuang8** [comment]
Thanks @MinniMinMin!
@Sraccity Ready for review :))

**Review by Sraccity — CHANGES_REQUESTED** [review]
# *Nitpick*: Requesting Visual Changes 

**Kudos**: On getting the PR up  @MinniMinMin 

## **Bug** (Visual): Menu button color 

"Menu" Button doesn't change according to the system's color. 

> (If changes take too long, just let me know. Try to focus on other subjects work)

image

**Comment by MinniMinMin** [comment]
> # _Nitpick_: Requesting Visual Changes
> **Kudos**: On getting the PR up @MinniMinMin
> 
> ## **Bug** (Visual): Menu button color
> "Menu" Button doesn't change according to the system's color.
> 
> > (If changes take too long, just let me know. Try to focus on other subjects work)
> 
> image

Hey @Sraccity, the code should now be updated so the menu button follows the system theme, LMK if there's more changes required/requested, TY!

image

image

File: src/styles/App.scss [hunk 1]
```diff
@@ -122,7 +124,20 @@ $dragBarWidth: 10px;
 // Left Dragbar
 #leftdragbar {
 	grid-area: leftdragbar;
+	width: 0px;
+	/* 或者您想要的宽度 */
```
Comment (Sraccity) [inline src/styles/App.scss:128]: Translation: or the width you want

File: src/styles/App.scss [hunk 2]
```diff
@@ -122,7 +124,20 @@ $dragBarWidth: 10px;
...
+	cursor: default;
+}
+
+#leftdragbar.collapsed {
+	width: 5px;
+}
 
+#leftdragbar .handle {
+	display: none;
+	/* 隐藏拖动手柄 */
```
Comment (Sraccity) [inline src/styles/App.scss:140]: Translation: Hide drag handles

File: src/styles/App.scss [hunk 3]
```diff
@@ -142,6 +157,44 @@ $dragBarWidth: 10px;
 
 
 
+#tabpages.full-width {
+	top: 60px;
+	/* 调整这个值以匹配新的 header 高度 */
+	left: 0;
+	width: calc(100% - 300px);
+	/* 假设右侧面板宽度为 300px，根据实际情况调整 */
```
Comment (Sraccity) [inline src/styles/App.scss:165]: Translations:

/* Adjust this value to match the new header height */
left: 0;
width: calc(100% - 300px);
/* Assume that the width of the right panel is 300px, adjust it according to the actual situation */

**Review by Sraccity — APPROVED** [review]
Well done. Imma merge it through. 

Cheers!

Some comments are in Mandarin but it is aite. Imma change it later*

(* Hopefully)

**Pull request merged**